google_directory_service.py

```diff
@@ -72,3 +73,21 @@ def GetUsersByGroupKey(self, group_key):
 
     return users
 
+  def SearchForUser(self, user_key):
+    """Search for a user based on a user key.
+
+    Args:
+      user_key: A string identifying an individual user.
+
+    Returns:
+      users: A list with that user in it or empty.
+      List format is used here for consistency with the other methods.
+    """
+    users = []
+    request = self.service.users().get(userKey=user_key, projection='full')
+    result = request.execute(num_retries=NUM_RETRIES)
+    if result['primaryEmail']:
+      users.append(result)
+
+    return users
```

Comment:
There was one already in the docstring, but I moved it up and expanded it slightly.